async-nats/src/jetstream/stream.rs

```diff
@@ -1262,3 +1232,120 @@ pub struct External {
     #[serde(rename = "deliver", skip_serializing_if = "is_default")]
     pub delivery_prefix: Option<String>,
 }
+
+use std::marker::PhantomData;
+
+#[derive(Debug, Default)]
+pub struct Yes;
+#[derive(Debug, Default)]
+pub struct No;
+
+pub trait ToAssign: Debug {}
+
+impl ToAssign for Yes {}
+impl ToAssign for No {}
+
+#[derive(Debug)]
+pub struct PurgeFuture<SEQUENCE, KEEP>
+where
+    SEQUENCE: ToAssign,
+    KEEP: ToAssign,
+{
+    stream: crate::jetstream::stream::Stream,
+    inner: PurgeRequest,
+    sequence_set: PhantomData<SEQUENCE>,
+    keep_set: PhantomData<KEEP>,
+}
+
+impl<SEQUENCE, KEEP> PurgeFuture<SEQUENCE, KEEP>
+where
+    SEQUENCE: ToAssign,
+    KEEP: ToAssign,
+{
+    /// Adds subject filter to [PurgeRequest]
+    pub fn filter<T: Into<String>>(mut self, filter: T) -> PurgeFuture<SEQUENCE, KEEP> {
+        self.inner.filter = Some(filter.into());
+        self
+    }
+}
+
+impl PurgeFuture<No, No> {
+    pub(crate) fn build(stream: Stream) -> PurgeFuture<No, No> {
```

Comment:
Subject is always required?

```suggestion
    pub(crate) fn from(stream: Stream, filter: String) -> PurgeFuture<No, No> {
```

Reply:
No. You can purge the whole stream.